src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1131,12 +1135,18 @@ public void operate() {
 
                             String destinationDir = store.getStoreDirPath() + File.separator + "version-"
                                             + Long.toString(pushVersion);
+
+                            store.setCurrentlyFetching(true);
```

Comment:
+1, let's make sure the check and the setting of this boolean flag is properly synchronized.